src/main/java/ru/springcourse/libraryApp2/service/BookService.java

```diff
@@ -39,32 +43,32 @@ public Page<Book> findBookByPhrase(String phrase, Pageable pageable) {
 
     @Transactional
     public void updateBook(int id, Book updatedBook) {
-        updatedBook.setId(id);
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new BookNotFoundException("Книга с таким id - " + id + " не найдена"));
+        book.setDescription(updatedBook.getDescription());
+        book.setTitle(updatedBook.getTitle());
         bookRepository.save(updatedBook);
     }
 
     @Transactional
-    public void saveNewBook(Book book) {
-        bookRepository.save(book);
+    public void saveNewBook(BookDTO book) {
+        Book newBook = bookMapper.map(book, Book.class);
+        bookRepository.save(newBook);
     }
 
     @Transactional
-    public boolean deleteBookByTitle(String title) {
+    public void deleteBookByTitle(String title) {
         Optional<Book> book = bookRepository.findByTitle(title);
         if (book.isPresent()) {
             bookRepository.delete(book.get());
-            return true;
-        }
-        return false;
+        } else throw new BookNotFoundException("Книга с таким названием - " + title + " не найдена");
```

Comment:
рекомендуется даже если одна строчка, указывать фигурные скобки)
```suggestion
        } else {
           throw new BookNotFoundException("Книга с таким названием - " + title + " не найдена");
        }
```